### PR TITLE
Extending the MTEXT entity to support 3d rotational values

### PR DESCRIPTION
**Pull request by AndrewJoubert**
Add the directional vector for the 3 dimensions.

Add a post scan calculation for the rotation between the different axis

### CODE REVIEW

File: lib/entities/mtext.js [hunk 1]
```diff
@@ -8,42 +8,65 @@ function EntityParser() {}
 EntityParser.ForEntityName = 'MTEXT';
 
 EntityParser.prototype.parseEntity = function(scanner, curr) {
-    var entity = { type: curr.value };
+    var initialEntity = {
+        type: curr.value
+    };
```
Comment (richnicholls404) [inline lib/entities/mtext.js]: fix your tabs/spaces

File: lib/entities/mtext.js [hunk 2]
```diff
@@ -8,42 +8,65 @@ function EntityParser() {}
...
             default:
-                helpers.checkCommonEntityProperties(entity, curr);
+                helpers.checkCommonEntityProperties(initialEntity, curr);
                 break;
         }
         curr = scanner.next();
     }
+
+    // Post scanning calculating of things
+    var entity = Object.assign(initialEntity, {
```
Comment (DavidRHulme) [inline lib/entities/mtext.js]: Maybe less nice, but if you do `entity.rotationXY = ` etc, then you wouldn't need to rename `entity` above

File: lib/entities/mtext.js [hunk 3]
```diff
@@ -8,42 +8,65 @@ function EntityParser() {}
...
+            :0,
+        // Return a YZ rotation if we have the direction vector values for z and y in degrees
+        rotationYZ: initialEntity.directionalVector
+            ? Math.atan(initialEntity.directionalVector.z/initialEntity.directionalVector.y) * (180/Math.PI)
+            :0,
+        // Return a ZX rotation if we have the direction vector values for x and z in degrees
+        rotationZX: initialEntity.directionalVector
+            ? Math.atan(initialEntity.directionalVector.x/initialEntity.directionalVector.z) * (180/Math.PI)
+            :0
+    });
```
Comment (DavidRHulme) [inline lib/entities/mtext.js]: Be sure you're following indentation style of the project

Comment (richnicholls404) [inline lib/entities/mtext.js]: This kind of functionality tends to be in helpers for this library. They would probably want it in one before merging a PR.

**Comment by DavidRHulme** [comment]
Maybe would we better for official slowplans account to fork, in case you leave! Or we can try and push for the original dxf guy to merge into his project.

**Pull request merged**